src/components/primitives/Button/ButtonGroup.tsx

```diff
@@ -27,41 +27,44 @@ export default memo(
       if (Array.isArray(children)) {
         computedChildren = React.Children.map(
           children,
-          (child: JSX.Element, index: number) => {
-            return React.cloneElement(child, {
-              key: `button-group-child-${index}`,
-              variant,
-              size,
-              colorScheme,
-              isDisabled,
+          (child: any, index: number) => {
+            if (child?.type?.displayName === 'Button') {
+              return React.cloneElement(child, {
+                key: `button-group-child-${index}`,
+                variant,
+                size,
+                colorScheme,
+                isDisabled,
 
-              // when buttons are attached, remove rounded corners of all buttons except extreme buttons
-              ...(isAttached ? { borderRadius: 0 } : {}),
-              ...(isAttached && index === 0
-                ? direction === 'column'
-                  ? { borderTopRadius: borderRadius }
-                  : { borderLeftRadius: borderRadius }
-                : {}),
-              ...(isAttached && index === children?.length - 1
-                ? direction === 'column'
-                  ? { borderBottomRadius: borderRadius }
-                  : { borderRightRadius: borderRadius }
-                : {}),
+                // when buttons are attached, remove rounded corners of all buttons except extreme buttons
+                ...(isAttached ? { borderRadius: 0 } : {}),
+                ...(isAttached && index === 0
```

Comment:
We need to take into consideration the possibility of the first child element being falsy. If that's the case then the button's corners will not be properly set.